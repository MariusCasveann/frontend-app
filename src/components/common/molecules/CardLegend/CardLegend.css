.row-rect {
    display: inline-block;
    margin-right: 10px;
    width: 35px;
    height: 15px;
}

.legend-container {
    max-height: 400px;
    align-content: flex-start;
    display: grid !important;
    grid-template-columns: 1fr 1fr;
}

.legend-container div.ant-typography {
    grid-template-columns: 1fr 8fr;
    align-items: start;
    margin-bottom: 0.2rem;
    margin-right: 1em;
}

/* Style for mobile phone */
@media only screen and (max-width: 1642px) {
    .legend-container {
        display: inline-table !important;
    }
}
