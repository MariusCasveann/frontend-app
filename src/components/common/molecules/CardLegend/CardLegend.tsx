import * as React from 'react';
import { Typography } from 'antd';
import './CardLegend.css';

interface CardLegendProps {
    labels: Array<string | string[]> | undefined;
    colors: string[];
}

export const CardLegend = ({ labels, colors }: CardLegendProps) => (
    <div className="legend-container">
        {labels &&
            labels.map((label: string | string[], index) => (
                <Typography.Paragraph key={index}>
                    <span className="row-rect" style={{ backgroundColor: `${colors[index]}` }} />
                    {label}
                </Typography.Paragraph>
            ))}
    </div>
);
