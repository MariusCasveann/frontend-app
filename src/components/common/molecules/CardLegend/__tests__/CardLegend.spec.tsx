import * as React from 'react';

import { CardLegend } from '../CardLegend';
import * as ShallowRenderer from 'react-test-renderer/shallow';

describe('CardLegend', () => {
    it('renders correctly', () => {
        const renderer = ShallowRenderer.createRenderer();
        renderer.render(<CardLegend labels={[]} colors={[]} />);
        const result = renderer.getRenderOutput();
        expect(result).toMatchSnapshot();
    });
});
