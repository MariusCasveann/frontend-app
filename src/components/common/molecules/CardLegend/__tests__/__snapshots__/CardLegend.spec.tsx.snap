// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardLegend renders correctly 1`] = `
<div
  className="legend-container"
/>
`;
